src/solve3.F

```diff
@@ -405,13 +406,12 @@ subroutine solve3(nstep,rbufsz,num_soil_layers,                  &
       if(timestats.ge.1) time_prsrho=time_prsrho+mytime()
 
 
-        call bcs_GPU(rr)
+      call bcs_GPU(rr)
+      call bcs_GPU(rf)
```

Comment:
I saw that you moved many subroutine calls like this and I did not think they were answer changing. I am just curious about why you want to re-order these calling sequences and how you determine to move their locations up or down?

Reply:
The moving of subroutine calls around was associated with adding the load-imbalance timers, and the desire to reduce the number of #ifdef _MPI directives in the code.  So I grouped the packing routines bc*_GPU and the MPI boundary exchange calls together.

Reply:
So does it mean that it does not matter if I call the `comm_*_GPU` subroutines before or after the `bc*_GPU` subroutines, as long as I group them together first?

Reply:
Jian,  I kept the overall order between bc*_GPU and comm_*_GPU calls.  The order is critical, it just does not matter if all the operations on variable 'foo' are performed before the operations on variable 'bar'.  For example  

Old order 

call bcs_GPU(foo)
call comm_start_GPU(foo)
call bcs_GPU(bar)
call comm_start_GPU(bar)

New order:

call bcs_GPU(foo)
call bcs_GPU(bar)
call comm_start_GPU(foo)
call comm_start_GPU(bar)

Reply:
Thanks @johnmauff . That is clear to me now.